### PR TITLE
Magento Release Notes 2 3 4 penultimate review copy

### PR DESCRIPTION
**Pull request by jfrontain**
This pull request (PR) adds the semi-final version of the 2.3.4 release notes. It includes all core issue descriptions and some technical review. It is missing descriptions for 14 community-contributed issue fixes.

### CODE REVIEW

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 1]
```diff
@@ -846,6 +890,28 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
...
+
+*  Order-related `save_after_commit` callbacks are now called for guest checkouts as expected. Previously, the corresponding `order _save_commit_after` events were not dispatched.
+
+<!--- MC-17332-->
+
+*  The product counter and page lister on **Catalog** > **Products** now works correctly after the **Add Store Code to Urls** setting has been enabled or disabled.
+
+<!--- MC-19250-->
+
+*  The new CONSUMERS_WAIT_FOR_MAX_MESSAGES environment variable lets merchants configure how consumers process messages from the message queue. You can set this variable by adding the `consumers-wait-for-messages` option to `bin magento queue:consumers:start`. Using this option can help Magento deployments avoid being affected by long delays in message queue processing. See [Start message queue consumers](https://devdocs.magento.com/guides/v2.3/config-guide/mq/manage-message-queues.html).
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: I thought the environment variable was for Cloud, while on-premise users specify a new argument?

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 2]
```diff
@@ -948,11 +1010,75 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
...
+
+*  The Admin Address Country drop-down list now takes its values from  the **Allow Countries**  setting that is configured for the Website Store View where the order was made.  *Fix submitted by Oleksii Lisovyi in pull request [24345](https://github.com/magento/magento2/pull/24345)*. [GitHub-12560](https://github.com/magento/magento2/issues/12560)
+
+<!--- ENGCOM-6043-->
+
+*  The Available Countries list  (**Stores** > **General**) has been updated to include the countries identified in the latest version of the Common Locale Data Repository (version  36). *Fix submitted by Bartłomiej Szubert in pull request [24927](https://github.com/magento/magento2/pull/24927)*. [GitHub-24710](https://github.com/magento/magento2/issues/24710)
+
+<!--- ENGCOM-5640-->
+
+*  Magento no longer serializes user data multiple times when data is loaded by the loadByUsername method. *Fix submitted by Jamie Saunders in pull request [23827](https://github.com/magento/magento2/pull/23827)*. [GitHub-23824](https://github.com/magento/magento2/issues/23824)
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]:
```suggestion
*  Magento no longer serializes user data multiple times when data is loaded by the `loadByUsername` method. *Fix submitted by Jamie Saunders in pull request [23827](https://github.com/magento/magento2/pull/23827)*. [GitHub-23824](https://github.com/magento/magento2/issues/23824)
```

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 3]
```diff
@@ -1020,6 +1146,20 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
 
 *  You can now successfully import an image from an external URL.
 
+### Index
+
+<!--- MC-18833-->
+
+*  The Tier Price REST API now considers account indexer mode as expected.
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]:
```suggestion
*  The `POST /V1/products/tier-prices` call now considers account indexer mode as expected.
```

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 4]
```diff
@@ -1034,6 +1174,50 @@ We have fixed hundreds of issues in the Magento 2.3.4 core code.
...
+
+*  You can now add products with custom options of all types to the shopping cart. *Fix submitted by Alexandr Skrashuk in pull request [25055](https://github.com/magento/magento2/pull/25055)*. [GitHub-24726](https://github.com/magento/magento2/issues/24726)
+
+<!--- ENGCOM-6186-->
+
+*  Decimal numbers have been added to the Sample File in Import CSV section. Previously, using the **Download Sample File** option on the product’s import CSV page created a sample CSV file with a weight of 1, which was of limited use. *Fix submitted by Cristian Sanclemente in pull request [25317](https://github.com/magento/magento2/pull/25317)*. [GitHub-23920](https://github.com/magento/magento2/issues/23920)
+
+<!--- ENGCOM-6171-->
+
+*  A deprecated method in `\Magento\MysqlMq\Model\Driver\Exchange` has been replaced. Previously, this method caused an exception. *Fix submitted by Tjitse in pull request [25289](https://github.com/magento/magento2/pull/25289)*. [GitHub-21904](https://github.com/magento/magento2/issues/21904)
```
Comment (keharper) [inline src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: You have to say which method was replaced!

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 5]
```diff
@@ -68,17 +68,17 @@ This release contains 250 enhancements to core quality, which improve the qualit
...
+*  **Integration with Adobe Stock image galleries**. The new bundled Adobe stock integration extension enables merchants to add high quality media assets to their website content without leaving the Magento Admin. Merchants can use the searchable interface in the Magento Media Gallery to explore, preview, license, and deploy stock images in website content. See [Adobe Stock Integration](https://docs.magento.com/m2/ee/user_guide/cms/adobe-stock.html) and [Using Adobe Stock Images](https://docs.magento.com/m2/ee/user_guide/cms/adobe-stock-manage.html).
 
 ### Page Builder
 
 Page Builder enhancements for this release include:
 
-*  Improved product sorting. Merchants can now sort by product position in category or list of product SKUs, and sort by defined parameters such as name or stock status.
+*  Improved product sorting. Merchants can now sort by product position in category or in list of producSKUs. Sorting by predefined parameters such as name or stock status remains available.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: replaced existing text with your suggestion

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 6]
```diff
@@ -68,17 +68,17 @@ This release contains 250 enhancements to core quality, which improve the qualit
...
 
-*  Improved product carousel. Merchants can choose how to showcase products in their content by selecting from predefined options in Page Builder Products content type.
+*  Product carousel. Merchants can choose how to showcase products in their content by selecting from predefined options in Page Builder Products content type: grid or carousel.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: added "the"

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 7]
```diff
@@ -68,17 +68,17 @@ This release contains 250 enhancements to core quality, which improve the qualit
...
 
-*  The content that merchants create in Page Builder is optimized for rendering on the storefront using the Venia Theme (PWA Studio). Unstructured content (HTML) that Page Builder generates and saves into a data base is converted into structured data that works in React and PWA Studio.
+*  Content created with Page Builder can now be displayed in storefronts built with PWA Studio using new PWA & Page Builder components. The unstructured content (HTML) that Page Builder generates and saves into a data base is converted into structured data that works in React and PWA Studio.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: got it

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 8]
```diff
@@ -116,7 +116,7 @@ For information on these enhancements plus other improvements, see [PWA Studio r
 
 ### dotdigital
 
-Live Chat powered by dotdigital enables merchants to increase conversion rates, and keep customers coming back with real-time engagement. All Magento 2.3.x merchants (both Magento Open Source and Magento Commerce) can receive a free live chat agent.
+Live Chat powered by dotdigital enables merchants to increase conversion rates, and keep customers coming back with real-time engagement. All Magento 2.3.x merchants (both Magento Open Source and Magento Commerce) can receive a free live chat agent without the need for a full dotdigital Engagement Cloud license.
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: got it

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md [hunk 9]
```diff
@@ -126,16 +126,14 @@ Live Chat powered by dotdigital enables merchants to increase conversion rates,
 
 New B2B features and improvements include the following:
 
-*  **Ability to export requisition lists into CSV format**. B2B buyers can then perform batch modification for subsequent import into the shopping cart or import back into the requisition list. <!--- MC-17417-->
+*  **Ability to export Requisition Lists into CSV format**. B2B buyers can export the contents of Requisition Lists for local editing. These exported lists can then be uploaded into B2B Quick Order for creation of new orders, quotes, or lists. See [Managing Requisition Lists](https://docs.magento.com/m2/b2b/user_guide/customers/account-dashboard-requisition-lists-manage.html).<!--- MC-17417-->
 
-*  **Granular ACLs for B2B modules**. Merchants can now restrict access to B2B features from the Admin,  controlling which  employees can work with B2B items and settings. <!--- MC-19649-->
+*  **Granular ACLs for B2B modules**. Merchants can now restrict access to B2B features from the Admin, controlling which employees can work with B2B items and settings. See [Role Resources](https://docs.magento.com/m2/ce/user_guide/system/permissions-role-resources.html. <!--- MC-19649-->
```
[the remark this answers is not in the document]
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: thank you for correcting that URL!

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md (hunk 1 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: This is the text that Margaret worked on with the developer, so I will not change it. Will follow up off-Git

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md (hunk 2 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: got it

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md (hunk 3 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: thank you for finding the correct call.

File: src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md (hunk 4 above)
Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: will ask community member

Comment (jfrontain) [reply src/guides/v2.3/release-notes/release-notes-2-3-4-commerce.md]: added the method name - thanks

**Comment by dobooth** [comment]
running tests

**Pull request merged**